activesupport/lib/active_support/core_ext/object/acts_like.rb

```diff
@@ -1,11 +1,32 @@
 # frozen_string_literal: true
 
 class Object
-  # A duck-type assistant method. For example, Active Support extends Date
-  # to define an <tt>acts_like_date?</tt> method, and extends Time to define
-  # <tt>acts_like_time?</tt>. As a result, we can do <tt>x.acts_like?(:time)</tt> and
-  # <tt>x.acts_like?(:date)</tt> to do duck-type-safe comparisons, since classes that
-  # we want to act like Time simply need to define an <tt>acts_like_time?</tt> method.
+  # Provides a way to check whether some class acts like some other class based on the existence of
+  # an appropriately-named marker method.
+  #
+  # A class that provides the same interface as <tt>X</tt> may define a marker method named
+  # <tt>acts_like_X?</tt> to signal its compatibility.
```

Comment:
Should this be lowercase instead? As we define `acts_like_time?` for `Time`?
```suggestion
  # <tt>acts_like_x?</tt> to signal its compatibility.
```
Or maybe use `SomeClass` instead of `X`?

Reply:
Agree! I've updated the PR.